DotNetHowToIndexers/Program.cs

```diff
@@ -0,0 +1,325 @@
+        /// </summary>
+        private static void DeleteIndexIfExists(SearchServiceClient serviceClient)
+        {
+            if (serviceClient.Indexes.Exists(IndexName))
+            {
+                serviceClient.Indexes.Delete(IndexName);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Azure Search index for the Hotel data type.
+        /// </summary>
+        private static void CreateIndex(SearchServiceClient serviceClient)
+        {
+            var definition = new Index()
+            {
+                Name = IndexName,
+                Fields = FieldBuilder.BuildForType<Hotel>()
+            };
+
+            serviceClient.Indexes.Create(definition);
+        }
+
+        /// <summary>
+        /// Deletes an Azure Search data source with the given name, if it exists.
+        /// </summary>
+        private static void DeleteDataSourceIfExists(SearchServiceClient serviceClient, string dataSourceName)
+        {
+            if (serviceClient.DataSources.Exists(dataSourceName))
+            {
+                serviceClient.DataSources.Delete(dataSourceName);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Azure Search indexer for the given data source with the given name.
+        /// </summary>
+        private static Indexer CreateIndexer(SearchServiceClient serviceClient, string dataSourceName, string indexerName)
+        {
+            Indexer indexer = new Indexer
+            {
+                DataSourceName = dataSourceName,
+                Name = indexerName,
+                Schedule = IndexerSchedule,
+                TargetIndexName = IndexName
+            };
+            return serviceClient.Indexers.Create(indexer);
+        }
+
+        /// <summary>
+        /// Deletes an Azure Search indexer if it exists.
+        /// </summary>
+        private static void DeleteIndexerIfExists(SearchServiceClient serviceClient, string indexerName)
+        {
+            if (serviceClient.Indexers.Exists(indexerName))
+            {
+                serviceClient.Indexers.Delete(indexerName);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Azure Search data source for a Azure SQL database.
+        /// </summary>
+        /// <remarks>
+        /// Requires a connection string be specified in the configuration file.
+        /// The table name is assumed to be "hotels" as specified in the accompanying
+        /// sql script.
+        /// </remarks>
+        private static DataSource CreateAzureSqlDataSource(SearchServiceClient serviceClient, IConfigurationRoot configuration)
+        {
+            DataSource azureSqlDataSource = DataSource.AzureSql(
+                name: AzureSqlDataSourceName,
+                sqlConnectionString: configuration[AzureSqlConnectionStringProperty],
+                tableOrViewName: AzureSqlTableName,
+                changeDetectionPolicy: new HighWaterMarkChangeDetectionPolicy
```

Comment:
Perhaps this would be a great place to make use of the `SqlIntegratedChangeTrackingPolicy` class?

And use the `HighWaterMarkChangeDetectionPolicy` for any of the other data sources?

Reply:
+1 to SqlIntegratedChangeTrackingPolicy

Reply:
@arv100kri there is no `HighWaterMarkChangeDetectionPolicy` for `DocumentDb` or `AzureTableStorage`. `DocumentDb` has a default `useChangeDetection=true`, but you can't control how it works (at least from the SDK https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.search.models.datasource.documentdb?view=azuresearch-3.0.4#Microsoft_Azure_Search_Models_DataSource_DocumentDb_System_String_System_String_System_String_System_String_System_Boolean_Microsoft_Azure_Search_Models_DataDeletionDetectionPolicy_System_String_).
`AzureTableStorage` doesn't appear to have any change detection option from the SDK. See https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.search.models.datasource.azuretablestorage?view=azuresearch-3.0.4